Sources/UIComponents/Molecules/NotificationBubbleView.swift

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+extension Components.Molecules {
+    open class NotificationBubbleView: DebugOverlayableView, Component {
+
+        private struct Consts {
+            static let defaultSize: CGSize = .init(width: 48, height: 24)
+        }
+
+        public enum NotificationMode {
+            /// Specifies that the notification will display a number in the notification bubble.
+            case number(count: Int, max: Int = 99, hideWhenZero: Bool = true)
+
+            /// Specifies that the notification will display the notification bubble alone
+            case circle
+
+            /// Specifies that the notification bubble will be hidden
+            case hidden
+        }
+
+        public struct Model {
+            public let notificationMode: NotificationMode
+
+            public init(notificationMode: NotificationMode) {
+                self.notificationMode = notificationMode
+            }
+        }
+
+        public lazy var countLabel: UILabel = {
```

Comment:
There's a new builder to simply this stuff now - see UIView+Build - it does the translatesAutoresizingMaskIntoConstraints for you so this would become -

countLabel = UILabel.build {
    $0.setAppearance(for: .bold)
    etc...
}

Reply:
Updated in 8619a1e

Reply:
👍